probability/distributions.cpp

```diff
@@ -0,0 +1,311 @@
+/**
+ * \file
+ * \brief Implementation of Discrete & Continuous Distributions
+ * \author [Tuan Ho](https://github.com/ttuanho)
+ * \note The implementation is based on [this
+ * cheatsheet](https://web.cs.elte.hu/~mesti/valszam/kepletek)
+ * \todo
+ * <ul>
```

Comment:
This is good documentation. you can also use the simple syntax of Markdown to format the documentation just like you used for the links.